certification/engine/engine.go

```diff
@@ -40,9 +40,17 @@ func NewForConfig(config runtime.Config) (CheckEngine, error) {
 		checks[i] = check
 	}
 
-	engine := &shell.CheckEngine{
-		Image:  config.Image,
-		Checks: checks,
+	var engine CheckEngine
+	if config.Mounted {
+		engine = &shell.MountedCheckEngine{
+			Image: config.Image,
+			Check: checks[0],
+		}
+	} else {
+		engine = &shell.CheckEngine{
+			Image:  config.Image,
+			Checks: checks,
+		}
```

Comment:
Nit: can we avoid this `else` here? Instead - just assign shell.CheckEngine with the values we want to `engine` in line 43, then your if block will change that if it needs to do so.

Reply:
Will do.